src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CsvDataExtractionService.kt

```diff
@@ -0,0 +1,102 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.CsvMapperConfig.csvMapper
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity.VideoBookingHistory
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.Location
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBookingHistoryRepository
+import java.io.OutputStream
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.stream.Stream
+import kotlin.streams.asSequence
+
+@Service
+class CsvDataExtractionService(
+  private val videoBookingHistoryRepository: VideoBookingHistoryRepository,
+  private val locationsService: LocationsService,
+) {
+
+  @Transactional(readOnly = true)
+  fun courtBookingsByHearingDateToCsv(fromDate: LocalDate, toDate: LocalDate, csvOutputStream: OutputStream) {
+    writeCourtBookingsToCsv(videoBookingHistoryRepository.findByMainDateBetween(fromDate, toDate), csvOutputStream)
+  }
+
+  @Transactional(readOnly = true)
+  fun courtBookingsByBookingDateToCsv(fromDate: LocalDate, toDate: LocalDate, csvOutputStream: OutputStream) {
+    writeCourtBookingsToCsv(videoBookingHistoryRepository.findByDateOfBookingBetween(fromDate, toDate), csvOutputStream)
+  }
+
+  private fun writeCourtBookingsToCsv(bookings: Stream<VideoBookingHistory>, csvOutputStream: OutputStream) {
+    val locationsByPrisonCode = mutableMapOf<String, List<Location>>()
+
+    val courtBookings = bookings.filter(VideoBookingHistory::isCourtBooking)
+      .peek { locationsByPrisonCode.getOrPut(it.prisonCode) { locationsService.getVideoLinkLocationsAtPrison(it.prisonCode, false) } }
+      .map { CourtBookingDto(it, locationsByPrisonCode) }
+      .asSequence()
+
+    csvMapper
+      .writer(csvMapper.schemaFor(CourtBookingDto::class.java).withHeader())
+      .writeValues(csvOutputStream.bufferedWriter())
+      .use { writer -> courtBookings.forEach(writer::write) }
+  }
+}
+
+@JsonPropertyOrder(
+  "timestamp",
+  "videoLinkBookingId",
+  "eventType",
+  "agencyId",
+  "court",
+  "courtId",
+  "madeByTheCourt",
+  "mainStartTime",
+  "mainEndTime",
+  "preStartTime",
+  "preEndTime",
+  "postStartTime",
+  "postEndTime",
+  "mainLocationName",
+  "preLocationName",
+  "postLocationName",
+)
+data class CourtBookingDto(
```

Comment:
Old BVLS has an event ID.  We don't.  Does this really matter?  We could use the appointment history id ...